lib/mongo/crypt/handle.rb

```diff
@@ -70,13 +79,29 @@ def initialize(kms_providers, kms_tls_options, options={})
         @bypass_query_analysis = options[:bypass_query_analysis]
         set_bypass_query_analysis if @bypass_query_analysis
 
+        @crypt_shared_lib_path = options[:crypt_shared_lib_path]
+        @explicit_encryption_only = options[:explicit_encryption_only]
+        if @crypt_shared_lib_path
+          Binding.setopt_set_crypt_shared_lib_path_override(self, @crypt_shared_lib_path)
+        elsif !@bypass_query_analysis && !@explicit_encryption_only
+          Binding.setopt_append_crypt_shared_lib_search_path(self, "$SYSTEM")
+        end
+
         @logger = options[:logger]
         set_logger_callback if @logger
 
         set_crypto_hooks
 
         Binding.setopt_kms_providers(self, kms_providers.to_document)
+
         initialize_mongocrypt
+
+        @crypt_shared_lib_required = !!options[:crypt_shared_lib_required]
+        if @crypt_shared_lib_required && crypt_shared_lib_version == 0
+          raise Mongo::Error::CryptError.new(
+            "Crypt shared library is required, but cannot be loaded"
```

Comment:
```suggestion
            "Crypt shared library is required, but cannot be loaded according to libmongocrypt"
```

Reply:
I assume libmongocrypt doesn't provide any diagnostics, if it does they should be incorporated in this exception message.

Reply:
Fixed.